src/web/browser/cookie.ts

```diff
@@ -6,6 +6,11 @@ const isValidCookieValue = (name: string): boolean =>
 
 const getShortDomain = (isCrossSubdomain = false): string => {
     const domain = document.domain || '';
+
+    if (domain === 'localhost') {
```

Comment:
This addition to `cookie.ts` fixes a bug where the consent Cookie was never successfully set on `localhost`, the same issue was recently(ish) fixed on `frontend`: https://github.com/guardian/frontend/pull/21650

Reply:
I tested this locally and yep, it now dismisses correctly